InternalPythonModules/android/viber.py

```diff
@@ -117,13 +117,29 @@ def parse_contacts(self, contacts_db, helper):
         try:
             contacts_parser = ViberContactsParser(contacts_db)
             while contacts_parser.next():
-                helper.addContact( 
-                    contacts_parser.get_contact_name(), 
-                    contacts_parser.get_phone(),
-                    contacts_parser.get_home_phone(),
-                    contacts_parser.get_mobile_phone(),
-                    contacts_parser.get_email()
-                )
+                if (not(not contacts_parser.get_phone() or contacts_parser.get_phone().isspace())):
+                    helper.addContact( 
+                        contacts_parser.get_contact_name(), 
+                        contacts_parser.get_phone(),
+                        contacts_parser.get_home_phone(),
+                        contacts_parser.get_mobile_phone(),
+                        contacts_parser.get_email()
+                    )
+                # Check if contact_name is blank and if it is not create a TSK_CONTACT otherwise ignore as not Contact Info
+                elif (not(not contacts_parser.get_contact_name() or contacts_parser.get_contact_name().isspace())):
+                    current_case = Case.getCurrentCase().getSleuthkitCase()
+                    attributes = ArrayList()
+                    artifact = contacts_db.getDBFile().newArtifact(BlackboardArtifact.ARTIFACT_TYPE.TSK_CONTACT)
+                    attributes.add(BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_NAME.getTypeID(), self._PARSER_NAME, contacts_parser.get_contact_name()))
+                    artifact.addAttributes(attributes)
+                    
+                    try:
+                    # Post the artifact to blackboard
+                       current_case.getBlackboard().postArtifact(artifact, self._PARSER_NAME)
+                    except Blackboard.BlackboardException as e:
```

Comment:
I think we should use the catch block around the while loop or, in other words, just omit the block here.